models/repo_redirect.go

```diff
@@ -38,7 +38,10 @@ func NewRepoRedirect(ownerID, repoID int64, oldRepoName, newRepoName string) err
 	}
 
 	if err := deleteRepoRedirect(sess, ownerID, newRepoName); err != nil {
-		sess.Rollback()
+		err = sess.Rollback()
+		if err != nil {
```

Comment:
This if statement is not necessary because no matter the outcome err will be returned